### PR TITLE
add harcoded authorizer type for well-known rules

### PR DESCRIPTION
**Pull request by deads2k**
Notably, we want to allow the metrics scraper since that rule is always present in openshift clusters

### CODE REVIEW

File: pkg/authorization/hardcodedauthorizer/hardcoded_authorizer.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	ret, err := NewHardcodedAuthorizer(rules, usernames...)
+	if err != nil {
+		panic(err)
+	}
+	return ret
+}
+
+// NewHardcodedMetricsScaperAuthorizer provides an authorizer that allows metrics scraping without contacting the
+// kube-apiserver for an authorization check on a rule we always have in place.
+func NewHardcodedMetricsScaperAuthorizer() authorizer.Authorizer {
```
Comment (deads2k) [inline pkg/authorization/hardcodedauthorizer/hardcoded_authorizer.go:76]: @s-urbaniak please confirm this rule and serviceaccount

**Comment by sttts** [comment]
What is this? Fork of RBAC?

/hold

File: pkg/controller/controllercmd/builder.go [hunk 2]
```diff
@@ -9,6 +9,9 @@ import (
 	"sync"
 	"time"
 
+	"github.com/openshift/library-go/pkg/authorization/hardcodedauthorizer"
```
Comment (mfojtik) [inline pkg/controller/controllercmd/builder.go:12]: order

**Comment by deads2k** [comment]
replaced with approach in https://github.com/openshift/library-go/pull/998

**Pull request closed without merging**